### PR TITLE
add track footprint to basic search results

### PR DESCRIPTION
**Pull request by satellitestudiodesign**
- [X] Add map and track
- [X] Request track only when the result is visible on screen
- [X] Request tracks as binary to avoid eating into user quota
- [x] Highlight segments based on TransmissionsTimeline hover
- [x] Show "something" when track response is empty

### CODE REVIEW

File: apps/fishing-map/features/search/TrackFootprint.tsx [hunk 1]
```diff
@@ -0,0 +1,132 @@
...
+  const onScreen = useOnScreen(fullCanvasRef)
+  const fullContext = fullCanvasRef.current?.getContext('2d')
+  const highlightContext = highlightCanvasRef.current?.getContext('2d')
+
+  const fetchData = async (vesselId: string) => {
+    if (!trackDatasetId) {
+      setError(true)
+      return
+    }
+    let data = await GFWAPI.fetch<any>(
```
Comment (j8seangel) [inline apps/fishing-map/features/search/TrackFootprint.tsx]: At some point, we should resolve this URL with the `resolveEndpoint` function, but this is enough for now to start testing

File: apps/fishing-map/hooks/screen.hooks.ts [hunk 2]
```diff
@@ -110,3 +110,28 @@ export const useDownloadDomElementAsImage = (
     previewImageLoading,
   }
 }
+
+export const useOnScreen = (ref, rootMargin = '0px') => {
```
Comment (j8seangel) [inline apps/fishing-map/hooks/screen.hooks.ts:114]: I'd say to move this to the reusable react-hooks library, as it looks very useful!

File: libs/react-hooks/src/use-small-screen/use-small-screen.ts [hunk 3]
```diff
@@ -3,7 +3,7 @@ import { useState, useEffect, useCallback } from 'react'
 const DEFAULT_BREAKPOINT = 768
 
 export function useSmallScreen(width = DEFAULT_BREAKPOINT) {
-  const [isSmallScreen, setIsSmallScreen] = useState<boolean>(false)
+  const [isSmallScreen, setIsSmallScreen] = useState<boolean>(true)
```
Comment (j8seangel) [inline libs/react-hooks/src/use-small-screen/use-small-screen.ts:6]: Isn't this a breaking change?

**Review by j8seangel — APPROVED** [review]
This is amazing 🥇 I love it!
I've left some minor comments and pushed a [commit](https://github.com/GlobalFishingWatch/frontend/commit/79d4e030faad468e71aa154a65720d9186de9056) with the reviewed changes.

File: libs/react-hooks/src/use-small-screen/use-small-screen.ts (hunk 3 above)
Comment (satellitestudiodesign) [reply libs/react-hooks/src/use-small-screen/use-small-screen.ts:6]: I checked where it was used and looked like `true` made more sense as the initial value in all cases. In some places it makes no difference but in others it avoids an unnecessary layout shift. We can review it together.

**Pull request merged**